### PR TITLE
Display unimplemented errors in Yul codegen testing

### PR DESCRIPTION
**Pull request by axic**
Based on #10235.

### CODE REVIEW

**Comment by axic** [comment]
@mijovic @chriseth I think this will be the final batch for the IR, the unimplemented features? :)

**Comment by axic** [comment]
So apparently nothing is triggered by tests?

File: test/libsolidity/SemanticTest.cpp [hunk 1]
```diff
@@ -178,16 +178,9 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 				constructed = true;
 			}
 		}
-		catch (langutil::UnimplementedFeatureError const&)
-		{
-			// ignore unimplemented feature errors when forcibly trying to compile via yul.
-			if (!_compileViaYul || m_runWithYul)
-				throw;
-		}
 		catch (langutil::Error const& _error)
```
Comment (axic) [inline test/libsolidity/SemanticTest.cpp]: Ah, this catches UnimplementedFeatureError too.

**Comment by mijovic** [comment]
Looks like nothing from tests is unimplemented :-)

**Comment by axic** [comment]
> Looks like nothing from tests is unimplemented :-)

We have more than 10 places with `solUnimplementedAssert`. Are these just not tested or this PR is broken?

**Comment by chriseth** [comment]
Some are also unimplemented in the old codegen (immutables with non-value types), some could maybe be triggered.

File: test/libsolidity/SemanticTest.cpp [hunk 2]
```diff
@@ -151,8 +151,6 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 
 	for (auto& test: m_tests)
 	{
-
-		try
 		{
```
Comment (chriseth) [inline test/libsolidity/SemanticTest.cpp:155]: Please remove and unindent.

**Review by chriseth — APPROVED** [review]

**Pull request merged**